src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -54,19 +52,32 @@ public static class AlternativeSavingCalculator
 
                 }
             }
-            var approximateSavedAmount = 0M;
+            var timesToRepeatSaving = 0;
+            var approximateSavedAmount = startingBalance;
             while (goal.Amount > approximateSavedAmount)
             {
                 for (var i = enumCount; i > (int)Importance.Necessary; i--)
                 {
-                    averagesOfAmountByImportanceAdjusted[i - 1] = sumsOfAmountByImportanceAdjusted[i - 1] / entryAmounts[i - 1];
-                    averagesOfAmountByImportanceDefault[i - 1] = sumsOfAmountByImportanceDefault[i - 1] / entryAmounts[i - 1];
+                    if (entryAmounts[i - 1] != 0)
+                    {
+                        averagesOfAmountByImportanceAdjusted[i - 1] = sumsOfAmountByImportanceAdjusted[i - 1] / entryAmounts[i - 1];
+                        averagesOfAmountByImportanceDefault[i - 1] = sumsOfAmountByImportanceDefault[i - 1] / entryAmounts[i - 1];
```

Comment:
Wait does these arrays just hold the same value or something like that. And if same operations in two lines, maybe lambda expression could be used here, but not sure, since it is pretty simple equation.

Reply:
I'll look into lambda expressions, but the arrays just hold the untouched values, values after they're adjusted and how many values there are in an importance, there's probably a better way to do this, but I went for simplicity and readability. Answer your question?